pkg/prowloader/prow.go

```diff
@@ -201,6 +203,39 @@ func (pl *ProwLoader) prowJobToJobRun(pj prow.ProwJob) error {
 	return nil
 }
 
+func (pl *ProwLoader) findOrAddPullRequests(refs *prow.Refs) []models.ProwPullRequest {
+	if refs == nil {
+		return nil
+	}
+	pulls := make([]models.ProwPullRequest, 0)
+
+	for _, pr := range refs.Pulls {
+		if pr.Link == "" {
+			continue
+		}
+
+		pull := models.ProwPullRequest{}
+		pl.dbc.DB.Where("link = ?", pr.Link).Find(&pull)
```

Comment:
Little bit confusing here, I don't see this ever getting created, so am I correct that we're relying on the fact that we try to find one with this link (you may want an index on link), if it exists we update it. If it doesn't exist, is this just working magically via not error checking, and getting an empty "pull" struct, then populating it, and saving it?

Neat if working but I'd go a little more explicit, could be some weirdness here if we got an error that wasn't "this doesn't exist" for example, and it might be helpful for future folks to see a little more clearly where we create vs update.

Reply:
It saves it down below. Save updates or creates.  I did it that way in case the PR title changes, we'll end up with the latest one.

Reply:
I see what you're saying though, I can fix it to be more explicit.